search-service/src/test/kotlin/com/egm/stellio/search/service/AttributeInstanceServiceTests.kt

```diff
@@ -80,15 +81,18 @@ class AttributeInstanceServiceTests {
         )
         attributeInstanceService.create(observation).block()
 
-        val temporalQuery = TemporalQuery(emptyList(), TemporalQuery.Timerel.AFTER, ZonedDateTime.now(ZONE_OFFSET).minusHours(1), null, null, null)
+        val temporalQuery = TemporalQuery(emptyList(), TemporalQuery.Timerel.AFTER, Instant.now().atZone(ZoneOffset.UTC).minusHours(1), null, null, null)
         val enrichedEntity = attributeInstanceService.search(temporalQuery, temporalEntityAttribute)
 
         StepVerifier.create(enrichedEntity)
             .expectNextMatches {
                 it.size == 1 &&
                     it[0]["attribute_name"] == "incoming" &&
                     it[0]["value"] == 12.4 &&
-                    ZonedDateTime.parse(it[0]["observed_at"].toString()) == observationDateTime &&
+                    // Postgres stores the observedAt value in UTC.
+                    // when retrieved, it's converted to the current timezone using the system variable timezone and with the +/- annotation.
+                    // For this reason, a cast to Z format and a cast to UTC is needed.
+                    ZonedDateTime.parse(it[0]["observed_at"].toString()).toInstant().atZone(ZoneOffset.UTC) == observationDateTime &&
```

Comment:
@bobeal, how can we deal with that ?
for reference: https://docs.postgresql.fr/10/datatype-datetime.html (8.5.1.3. Horodatage)

Reply:
At this point, this is not really important as data is still "internal". What is important is that consumers receive the dates in the excepted format.